pkg/operation/botanist/component/coredns/coredns.go

```diff
@@ -526,6 +533,137 @@ import custom/*.server
 				},
 			},
 		}
+
+		clusterProportionalDNSAutoscalerServiceAccount = &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "kube-dns-autoscaler",
+				Namespace: metav1.NamespaceSystem,
+			},
```

Comment:
Set `automountServiceAccountToken=false`

Reply:
Why should the service account token not by automounted? The autoscaler needs access to the `kube-apiserver` to get the cluster size in terms of nodes and scale CoreDNS accordingly.
In other words: It does not seem to work if I set `automountServiceAccountToken=false`, but crashes instead:
```
E0125 10:26:47.318985       1 autoscaler.go:52] open /var/run/secrets/kubernetes.io/serviceaccount/token: no such file or directory
```

Reply:
Setting it to `false` should trigger the projected token mount (https://github.com/gardener/gardener/blob/master/docs/concepts/resource-manager.md#auto-mounting-projected-serviceaccount-tokens) and static token invalidation). Did you rebase to `master` when testing this?

Reply:
I just rebased and set everything up according to the link you provided, i.e. added `automountServiceAccountToken=false` to the deployment as well as the service account, but it does not work for me :-( .

Reply:
Updated with newest commit.